packages/datadog-plugin-jest/src/index.js

```diff
@@ -0,0 +1,167 @@
+}
+
+function getEnvironment (BaseEnvironment) {
+  return class DatadogJestEnvironment extends BaseEnvironment {
+    constructor (config, context) {
+      super(config, context)
+      this.testSuite = context.testPath.replace(`${config.rootDir}/`, '')
+      this.rootDir = config.rootDir
+    }
+    async setup () {
+      if (!global.tracer) {
+        const ciMetadata = getCIMetadata()
+        const { repository, branch, commit } = await getGitInformation()
+        global.tracer = require('../../dd-trace').init({
+          sampleRate: 1,
+          flushInterval: 1,
+          startupLogs: false,
+          ingestion: {
+            sampleRate: 1
+          },
+          tags: {
+            ...ciMetadata,
+            [GIT_COMMIT_SHA]: commit,
+            [GIT_BRANCH]: branch,
+            [GIT_REPOSITORY_URL]: repository,
+            [BUILD_SOURCE_ROOT]: this.rootDir,
+            [TEST_FRAMEWORK]: 'jest'
+          }
+        })
+      }
+
+      return super.setup()
+    }
+    async teardown () {
+      await new Promise((resolve) => {
+        global.tracer._tracer._exporter._writer.flush(resolve)
+      })
+      return super.teardown()
+    }
+
+    async handleTestEvent (event) {
+      if (event.name === 'test_skip' || event.name === 'test_todo') {
+        global.tracer.startSpan(
+          'jest.test',
+          {
+            tags: {
+              // Since we are using `startSpan` we can't use `type` and `resource` options
+              // so we have to manually set the tags.
+              [SPAN_TYPE]: 'test',
+              [RESOURCE_NAME]: `${event.test.parent.name}.${event.test.name}`,
+              [TEST_TYPE]: 'test',
+              [TEST_NAME]: event.test.name,
+              [TEST_SUITE]: this.testSuite,
+              [TEST_STATUS]: 'skip'
+            }
+          }
+        ).finish()
+      }
+      if (event.name === 'test_start') {
+        let specFunction = event.test.fn
+        if (specFunction.length) {
+          specFunction = promisify(specFunction)
+        }
+        event.test.fn = global.tracer.wrap(
+          'jest.test',
+          { type: 'test',
+            resource: `${event.test.parent.name}.${event.test.name}`,
+            tags: {
+              [TEST_TYPE]: 'test',
+              [TEST_NAME]: event.test.name,
+              [TEST_SUITE]: this.testSuite
+            } },
+          async () => {
+            let result
+            try {
+              result = await specFunction()
+              setTestStatus('pass')
+            } catch (error) {
+              setTestStatus('fail')
+              throw error
+            } finally {
+              finishStartedSpans()
+            }
+            return result
+          }
+        )
+      }
+    }
+  }
+}
+
+module.exports = {
+  name: 'jest',
+  // ** Important: This needs to be the same as the versions for datadog-plugin-jest-circus
+  versions: ['>=24.8.0'],
+  getEnvironment
```

Comment:
Where are the `patch` and `unpatch` functions? Not sure I understand how this works since the instrumenter doesn't support `getEnvironment`.

Reply:
yeah this is a special case. I kind of adjusted to the format but it is not supposed to be used in the same way. 

What we end up providing is a wrapper for the `testEnvironment` in `jest`. In your test setup you need to import this and wrap whatever environment you're using. 

At first I was just exporting `getEnvironment` but then the testing setup was not working. 

Should we just bail out of this format and fix testing in a different way?